core/server/6elementUtils.js

```diff
@@ -0,0 +1,98 @@
+"use strict";
+
+var sixSenseDecoder = require('6sense/src/codec/decodeFromSMS.js');
+var genericCodec = require('quipu/parser.js');
+require('es6-shim');
+
+function getClientName(client) {
+
+	return (client === undefined ? undefined : client.name);
+}
+
+function getDataType(data) {
+	if (data.toString().match(/^net(\d)$/))
+		return 'network'
+	else if (data.toString() === 'timeout?' || data.toString().match("name=*"))
+		return 'request'
+	else if (data.toString().slice(0, 1) === '0')
+		return 'message'
+	else if (data.toString().slice(0, 1) === '1')
+		return 'data'
+	else if (data.toString().slice(0, 1) === '2')
+		return 'status'
+	else
+		return 'other'
+}
+
+function printMsg(msg, client) {
+	return new Promise(function(resolve) {
+		decode(msg)
+		.then(function(decoded) {
+			var type = getDataType(msg);
+			var clientName = getClientName(client);
+			switch (type) {
+				case 'network':
+					console.log('['+clientName+']'+"[NETWORK]>" + decoded.toString());
+					break;
+				case 'request':
+					console.log('['+clientName+']'+"[REQUEST]>" + decoded.toString());
+					break;
+				case 'message':
+					console.log('['+clientName+']'+"[MESSAGE]>" + decoded.toString());
+					break;
+				case 'data':
+					console.log('['+clientName+']'+"[DATA]   >" + decoded.toString());
+					break;
+				case 'status':
+					console.log('['+clientName+']'+"[STATUS] >" + decoded.toString());
+					break;
+				default:
+					console.log('['+clientName+']'+"[OTHER]  >" + decoded.toString());
+					break;
+			}
+
+			resolve({decoded: decoded.toString(), type: type});
+		})
+	});
+}
+
+// Decode any message received by SMS or TCP
+function decode(message) {
+
+	return new Promise(function(resolve){
+
+		switch ('012'.indexOf(message.toString().charAt(0))) {
```

Comment:
This should probably read `message[0]` (message is a string, isn't it?) and the -1 case should be handled by a `default: reject(new Error('unexpected message type '+message[0]))`